configure.ac

```diff
@@ -685,6 +685,11 @@ CXX="$ac_save_CXX"
 
 TS_ADDTO(AM_CXXFLAGS, [-std=c++17])
 
+dnl find a most recent version of std::[experimental::]filesystem (c++fs or stdc++fs) available assuming c++17 and above
+AC_CHECK_LIB([c++fs], [main], ,
+    [ AC_CHECK_LIB([stdc++fs], [main], , , -lstdc++fs) ],
```

Comment:
I think @SolidWallOfCode tried this before, and we gave up on it.  I could not find support for this that worked universally on all our platforms.

Reply:
@zwoop, chatted with @SolidWallOfCode offline at the ATS summit and also looked into PR #4206 and `ts::file`. 

The idea of PR #4206 seems reasonable to me but the problem is that `ts::file` functionality is pretty limited at this moment and to use it we would need to keep enhancing it quite a bit and make sure it works on all supported platforms (i.e. this PR uses much more then just `ts::file::path`). Since in general we encourage newer C++ compilers and libraries (i.e. C+17 is minimum requirement now), I wonder if all that work is worth doing only to throw it all away soon (this was mentioned in the PR #4206 considerations as well). 

There are some functional differences in the `Filesystem` functionality between experimental and non-experimental versions (although `path /` is the only one I have noticed so far) but I wonder if we could discover and handle such differences in the our code as we go on case by case basis since the disappearance of the experimental version doesn't seem too far in the future.

Looking at our CI it seems `CentOS`, `Ubuntu`, `Debian` and `OSX` builds are fine and I was able to build on `FreeBSD 11.2` using `gcc7` (with no changes) and `clang6` (had to enable `c++experimental` by building a custom version  of `libc++`). I was curious to know what other platforms + compilers versions are failing so I can give them a try.